File: javascript/recursion/5-PowerOfInteger.js
// Write a recursive function to calculate the power of an integer raised to an exponent.

function power(base, exponent) {
  if (exponent === 0) {
    return 1;
  } else {
    return base * power(base, exponent - 1);
  }
}

// Example usage
const base = 3;
const exp = 4;
console.log(`${base} raised to the power of ${exp} is:`, power(base, exp)); // Expected output: 81

File: python/recursion/5-PowerOfInteger.py
# Write a recursive function to calculate the power of 
# an integer raised to an exponent.

def power(base, exponent):
  # Base case: exponent is 0, return 1
  if exponent == 0:
    return 1
  # Recursive case: multiply base with
  # power of (base, exponent - 1)
  return base * power(base, exponent - 1)

# Test the power function
print("3 to the power of 4 is equal to:", power(3, 4))